src/MongoDB/WriteConcern.c

```diff
@@ -358,7 +368,7 @@ static HashTable* php_phongo_write_concern_get_properties_hash(zval* object, boo
 		if (wtimeout != 0) {
 			zval z_wtimeout;
 
-			ZVAL_LONG(&z_wtimeout, wtimeout);
+			php_phongo_int64_to_zval(wtimeout, &z_wtimeout, is_bson TSRMLS_CC);
```

Comment:
Does it make sense to always call `php_phongo_int64_to_zval` here? `php_phongo_write_concern_get_properties_hash` is called from three places:

 * `get_debug_info` handler
 * `get_properties` handler
 * `bsonSerialize()` method

We know that `bsonSerialize()` needs to conditionally use an Int64 object if the value exceeds the range of a 32-bit integer. If `wtimeout` is within the range of a 32-bit integer, we can add it as an integer primitive.

The debug and properties handlers need not ensure compatibility between 32-bit and 64-bit architectures (unlike `serialize()`, so I think they'd do well to also encode as an integer whenever possible and only fall back to a string on a 32-bit architecture when `wtimeout` exceeds the range of a 32-bit integer.

Reply:
The idea was to always export as a string, since code might be serialised using `var_export` on a 64-bit architecture and then read on a 32-bit architecture. In this case, we'd export as an int, but then be unable to read it. That was the main reason for always treating it as if we were on a 32-bit platform.

Reply:
Ah, I wasn't too concerned about `var_export` being used across platforms since it actually produces code that the user would probably manually edit (unless they're using `eval()`). I was thinking of `serialize()` as the more likely way folks might transport between platforms.

In my experience (from GitHub issues), most people using `var_export` are doing so as an alternative to `var_dump` and didn't actually rely on `__set_state()` to rebuild an object.

I'm happy to have `var_export` produce a string as needed, though.